### ISSUE TITLE
doubt about restartWithIntent  

### ISSUE THREAD

**Issue by hongyangAndroid** [JakeWharton/ProcessPhoenix#3]
Whether should the restartWithIntent add FLAG_ACTIVITY_NEW_TASK and FLAG_ACTIVITY_CLEAR_TASK ? 

if Activity A -> MainActivity , then click " Restart Process with Intent "(just like your demo) ;
the Activity stack won't be clear , even contains ProcessPhoenix .

**Comment by hongyangAndroid** [comment]
I have made a test ;
When Activity A -> MainActivity , then click " Restart Process with Intent "(just like your demo) ;
```xml
numActivities=3 

Activities=[
ActivityRecord{5296d284 u0 
   com.jakewharton.processphoenix.sample/.ActivityA t71}, 
ActivityRecord{529a5788 u0 com.jakewharton.processphoenix.sample/
   com.jakewharton.processphoenix.ProcessPhoenix t71}, 
ActivityRecord{529b1704 u0 
   com.jakewharton.processphoenix.sample/.MainActivity t71}]
```
then click back , the ProcessPhoenix will recreate and startActivity to MainActivity ...

===>

then , i change the `Runtime.getRuntime().exit(0)`  to finish() in onCreate method of  ProcessPhoenix , then it run normally ; 


hope for your reply~

**Comment by JakeWharton** [comment]
To be honest, I have never used `restartWithIntent`. I believe it was requested by someone else back when this project was just a gist.

When using the button in the sample app I notice that pressing back resumes other activities instead of only having one. I'll have a look at some point, but I would welcome a PR to fix the issue if someone is up for it.

**Comment by JakeWharton** [comment]
Fixed by #5
